### ISSUE TITLE
Some errors when ensemble models

### ISSUE THREAD

**Issue by Lg955** [ultralytics/yolov5#3075]
## 🐛 Bug
When using ensemble models, sometimes(ensemble yolo5x) get a error.

## To Reproduce (REQUIRED)

Input:
```
python detect.py --source test-A --weights runs/train/exp4_yolo5l6/weights/best.pt runs/train/exp5_yolo5x/weights/best.pt --view-img --conf-thres 0.000 --augment
```

Output:

In my 4 experiences(5x, 5x6+5l6, 5x6+5x, 5l6+5x), the first 2 are successful, others are failed.

- yolo5x:
![5x - 副本](https://user-images.githubusercontent.com/69025725/117522796-d8878b80-afe7-11eb-8e45-8fc1e1c9e6f9.png)

- yolo5x6+5l6:
![5x6+5l6 - 副本](https://user-images.githubusercontent.com/69025725/117522813-eb01c500-afe7-11eb-84c3-3a7451b1d358.png)

- yolo5x6+5x:
![5x6+5x - 副本](https://user-images.githubusercontent.com/69025725/117522840-079dfd00-afe8-11eb-8ce7-0f7fefe40e6f.png)

- yolo5l6+5x:
![5l6+5x - 副本](https://user-images.githubusercontent.com/69025725/117522829-0076ef00-afe8-11eb-9452-0e6fca9f44d9.png)

```
Ensemble created with ['runs/train/exp4_yolo5l6/weights/best.pt', 'runs/train/exp5_yolo5x/weights/best.pt']

image 1/1200 /dataset/lg_datacode/code/yolov5/Underwater_optics/test-A/000001.jpg: Traceback (most recent call last):
  File "detect.py", line 199, in <module>
    detect(opt=opt)
  File "detect.py", line 78, in detect
    pred = model(img, augment=opt.augment)[0]
  File "/home/lg/anaconda3/envs/yolo5/lib/python3.8/site-packages/torch/nn/modules/module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/dataset/lg_datacode/code/yolov5/models/experimental.py", line 106, in forward
    y.append(module(x, augment)[0])
  File "/home/lg/anaconda3/envs/yolo5/lib/python3.8/site-packages/torch/nn/modules/module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/dataset/lg_datacode/code/yolov5/models/yolo.py", line 113, in forward
    yi = self.forward_once(xi)[0]  # forward
  File "/dataset/lg_datacode/code/yolov5/models/yolo.py", line 139, in forward_once
    x = m(x)  # run
  File "/home/lg/anaconda3/envs/yolo5/lib/python3.8/site-packages/torch/nn/modules/module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/dataset/lg_datacode/code/yolov5/models/common.py", line 210, in forward
    return torch.cat(x, self.d)
RuntimeError: Sizes of tensors must match except in dimension 2. Got 23 and 24 (The offending index is 0)
```

## Expected behavior
I think there are some bugs in the code, but I can not find it.

## Environment

 - OS: Ubuntu 18.04
 - GPU 4*2080Ti
 - cuda 10.1
 - python 3.8.8
 - pytorch 1.7.0

**Comment by glenn-jocher** [comment]
@Lg955 thanks for the bug report! I am able to reproduce this using your example code, I see the same error. I have a few ideas for what might be causing this, either the dataloader input stride is not synced to the P6 model in the ensemble, or the ensemble module itself is not flattening the outputs into a single list. I'll debug later and try to get a fix pushed this weekend.

<img width="1370" alt="Screenshot 2021-05-08 at 10 38 47" src="https://user-images.githubusercontent.com/26833433/117532780-99f2d080-afe9-11eb-9929-5b757986c5f8.png">

**Comment by glenn-jocher** [comment]
@Lg955 short term workaround seems to be to always pass the P6 model last if you have P6 models in the ensemble:

<img width="1370" alt="Screenshot 2021-05-08 at 10 41 12" src="https://user-images.githubusercontent.com/26833433/117532874-f9e97700-afe9-11eb-86a5-976398502b76.png">

**Comment by Lg955** [comment]
> @Lg955 short term workaround seems to be to always pass the P6 model last if you have P6 models in the ensemble:
> 
> <img alt="Screenshot 2021-05-08 at 10 41 12" width="1370" src="https://user-images.githubusercontent.com/26833433/117532874-f9e97700-afe9-11eb-86a5-976398502b76.png">

Wow, very nice, I succeeded!!!   
Looking forward to your future fix.

**Comment by glenn-jocher** [comment]
@Lg955 good news 😃! Your original issue may now been fixed ✅ in PR #3082.  To receive this update you can:
- `git pull` from within your `yolov5/` directory
- `git clone https://github.com/ultralytics/yolov5` again
- Force-reload [PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/): `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- View our updated notebooks:  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
